docs: add NG0403.md for Module Bootstrapped Without Bootstrap Component (#48483)
angular/angular: aio/content/errors/NG0403.md

> @name Module Bootstrapped Without Bootstrap Component

Reviewer: ```suggestion
@name Bootstrapped NgModule doesn't specify which component to initialize
```

Author: done

@category runtime

> @shortDescription Module AppModule Was Bootstrapped, But It Does Not Declare "@NgModule.bootstrap" Components Nor A "ngDoBootstrap" Method Please define one of these.

Reviewer: ```suggestion
@shortDescription An NgModule that was used for bootstrapping does not specify which component should be initialized.
```

Author: @AndrewKushnir changes will not use this sentence at all

@description

> This error mean the Module declared to be bootstrapped in the `main.ts` file (usually the `AppModule`) has empty `bootstrap` array or the `bootstrap` property is missing altogether

Reviewer: ```suggestion
This error means that an NgModule that was used for bootstrapping an application is missing key information for Angular to proceed with the bootstrap process.

The error usually happens when the NgModule is missing the `bootstrap` property in the `@NgModule` annotation or there is no `ngDoBootstrap` lifecycle hook defined on that NgModule class. More information about the bootstrapping process can be found in [this guide](guide/bootstrapping).
```

Author: saying `or` means both should be presented, so I made it `and`

also added the possibility that the  `bootstrap` property having empty array is same as being missing, had to alter your sentence a bit to add it this info

> both of these examples will throw the error

Reviewer: ```suggestion
The following examples will trigged the error.
```

Author: done

```typescript

> @NgModule({
>   declarations: [AppComponent],
>   imports: [BrowserModule, AppRoutingModule],
>   providers: [],
> })
> export class AppModule {}

Reviewer: ```suggestion
@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, AppRoutingModule],
  providers: [],
})
export class AppModule {}

// The `AppModule` is used for bootstrapping, but the `@NgModule.bootstrap` field is missing.
platformBrowser().bootstrapModule(AppModule);
```

Author: done

```

```typescript

> @NgModule({
>   declarations: [AppComponent],
>   imports: [BrowserModule, AppRoutingModule],
>   providers: [],
>   bootstrap: [],
> })
> export class AppModule {}

Reviewer: ```suggestion
@NgModule({
  declarations: [AppComponent],
  imports: [BrowserModule, AppRoutingModule],
  providers: [],
  bootstrap: [],
})
export class AppModule {}

// The `AppModule` is used for bootstrapping, but the `@NgModule.bootstrap` field contains an empty array.
platformBrowser().bootstrapModule(AppModule);
```

Author: done

```

@debugging

> Did you mistakenly removed the `bootstrap` property of your `AppModule` or remove the `AppComponent` from it?

Reviewer 2: ```suggestion
Did you mistakenly remove the `bootstrap` property of your `AppModule` or remove the `AppComponent` from it?
```

Author: [AndrewKushnir](https://github.com/AndrewKushnir)

> @debugging
> Did you mistakenly removed the `bootstrap` property of your `AppModule` or remove the `AppComponent` from it?

Reviewer: ```suggestion
@debugging
Please make sure that the NgModule that is used for bootstrapping is setup correctly:
- either the `bootstrap` property exists (and contains a non-empty array) in the `@NgModule` annotation
- or the `ngDoBootstrap` method exists on the NgModule class

You can find additional information on the bootstrapping process in [this guide](guide/bootstrapping#the-bootstrap-array). 

```

Author: > You can find additional information on the bootstrapping process in [this guide](guide/bootstrapping#the-bootstrap-array).

is duplicate of same info given in the description, so I didn't add this part

> 

my pleasure to contribute
this change is included now
